pennylane/device_tracker.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from collections import defaultdict
+
+import pennylane as qml
+
+
+def track(dev, version="default", **kwargs):
+    r"""Creates a tracking context and applies it to a device.
+
+    Args:
+        dev (~.Device): a PennyLane-compatible device
+        version (str): name of tracker to use.  The current options are
+            `default` and `timing`.
+
+    Keyword Args:
+        reset_on_enter=True (bool): whether or not to reset information
+            entering the context
+
+    **Usage Information**
+
+    Note that with backpropagation, this functions should take ``qnode.device``
+    instead of the device used to create the QNode.
```

Comment:
```suggestion
    .. note::
        With backpropagation, this functions should take ``qnode.device``
        instead of the device used to create the QNode.
```
Could also move this to below line 55